components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/internal/ClientManager.java

```diff
@@ -112,8 +111,10 @@ private PoolingNHttpClientConnectionManager createPoolingConnectionManager() thr
 
         String maxConnectionsString = IdentityUtil.getProperty(CONNECTION_POOL_MAX_CONNECTIONS);
         String maxConnectionsPerRouteString = IdentityUtil.getProperty(CONNECTION_POOL_MAX_CONNECTIONS_PER_ROUTE);
-        int maxConnections = DEFAULT_MAX_CONNECTIONS;
-        int maxConnectionsPerRoute = DEFAULT_MAX_CONNECTIONS;
+        int maxConnections = WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration()
+                .getDefaultMaxConnections();
+        int maxConnectionsPerRoute = WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration()
+                .getDefaultMaxConnections();
 
         if (StringUtils.isNotEmpty(maxConnectionsString)) {
             try {
```

Comment:
get rid of these checks as well, since we read `maxConnections` and `maxConnectionsPerRoute` from the configs.